### PR TITLE
Implemented carousel

### PR DESCRIPTION
**Pull request by ssebexen**
Changed the project display from a grid to a carousel.

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 9653227fd396cf732336300c112f9c48b6d3b1ec  | 
|--------|--------|

### Summary:
Transformed project display into a carousel with enhanced navigation and interactive features for inactive project tiles, including updates to event handling and styling.

**Key points**:
- Updated `ProjectList.tsx` to implement carousel functionality.
- Added `useState` for `currentProjectIndex`.
- Created `wrapProjectIndex` function for index wrapping.
- Updated `ProjectListTileInactive.tsx` to handle click events with `MouseEventHandler`.
- Styled inactive projects with reduced opacity, different background, and to indicate interactivity.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

File: app/components/ProjectList.tsx [hunk 1]
```diff
@@ -10,9 +13,42 @@ interface ProjectListProps {
 }
 
 export default function ProjectList(props: ProjectListProps) {
+
+  const [currentProjectIndex, setCurrentProjectIndex] = useState(0);
```
Comment (SamuelRCrider) [inline app/components/ProjectList.tsx:17]: Remember to type your useState: `useState<number>(0)`

File: app/components/ProjectList.tsx [hunk 2]
```diff
@@ -10,9 +13,42 @@ interface ProjectListProps {
...
+    else if (index >= props.projectList.projects.length) {
+      return 0;
+    }
+    else {
+      return index;
+    }
+
+  }
+
+  function incrementProjectIndex() {
```
Comment (SamuelRCrider) [inline app/components/ProjectList.tsx:33]: you can also write functions like this: `const incrementProjectIndex = () => {}`

File: app/components/ProjectList.tsx (hunk 1 above)
Comment (ssebexen) [reply app/components/ProjectList.tsx:17]: Huh, thanks for shouting that out. I had absolutely no idea that functions could be typed like that.

File: app/components/ProjectList.tsx (hunk 2 above)
Comment (ssebexen) [reply app/components/ProjectList.tsx:33]: Is there a general guideline on when to use this kind of function declaration (besides when you need the closure)? As a habit I tend to only use arrow functions for anonymous functions, but I'm not sure what the common practice is in JavaScript.

**Pull request merged**